src/platform/errors/errorHandler.node.ts

```diff
@@ -53,9 +53,14 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
         @inject(IKernelDependencyService) private readonly kernelDependency: IKernelDependencyService,
         @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService
     ) {}
+    private handledErrors = new WeakSet<Error>();
     public async handleError(err: Error): Promise<void> {
         traceWarning('DataScience Error', err);
         err = WrappedError.unwrap(err);
+        if (this.handledErrors.has(err)) {
```

Comment:
We end using this in a few places, this ensures we don't handle the same error over & over.

Reply:
This fixes #9521